blog/2024-01-12-pulsar-2023-year-in-review.md

```diff
@@ -0,0 +1,136 @@
+- [Pulsar Go Client 0.11.1](https://github.com/apache/pulsar-client-go/releases/tag/v0.11.1)
+- [Pulsar Node.js Client 1.9.0](https://github.com/apache/pulsar-client-node/releases/tag/v1.9.0)
+- [Pulsar Python Client 3.3.0](https://github.com/apache/pulsar-client-python/releases/tag/v3.3.0)
+- [Pulsar Manager 0.4.0](https://github.com/apache/pulsar-manager/releases/tag/v0.4.0)
+- [Pulsar Helm Chart 3.1.0](https://github.com/apache/pulsar-helm-chart/releases/tag/pulsar-3.1.0)
+- [Pulsar dotnet Client 3.1.1](https://github.com/apache/pulsar-dotpulsar/blob/master/CHANGELOG.md#311---2023-12-11)
+- [Reactive Client for Apache Pulsar 0.1.0](https://github.com/apache/pulsar-client-reactive/releases/tag/v0.5.1)
+
+For more information, see the [Clients Release Notes page](pathname:///release-notes/clients/).
+
+## Pulsar ecosystem
+
+In 2023, the Pulsar community worked with other open-source communities to add more integrations to the Pulsar ecosystem. Notable integrations include:
+
+- [Quarkus Extension for Apache Pulsar](https://quarkus.io/guides/pulsar) provides support for Apache Pulsar through SmallRye Reactive Messaging framework. Based on Eclipse MicroProfile Reactive Messaging specification 3.0, it proposes a flexible programming model bridging CDI and event-driven.
+
+- [Spring for Apache Pulsar](https://spring.io/blog/2023/11/21/spring-for-apache-pulsar-1-0-0-goes-ga/) provides a `PulsarTemplate` for publishing to a Pulsar topic and a `PulsarListener` annotation for consuming from a Pulsar topic, as well as various convenience APIs for Spring developers to ramp up their development journey into Apache Pulsar. Support is also included in Spring Boot via [auto-configuration](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#messaging.pulsar).
+
+- [Oxia](https://github.com/streamnative/oxia): currently, the practical limit for the number of topics manageable by a single Pulsar cluster is around 1 million topics. The lack of horizontal scalability in Zookeeper is one of the reasons for this limit. [Oxia](https://github.com/streamnative/oxia), released this year, is a scalable metadata store and coordination system. Replacing Zookeeper with Oxia in a Pulsar cluster allows for exceeding this 1M topics limit, although it is not the only prerequisite. This contributes to the goal of reaching 100M topics.
+
+For more information, see the [Ecosystem page](pathname:///ecosystem/).
+
+# What’s next in 2024
+
+## Community Improvements
+The Pulsar community seeks to further improve the project from different aspects. 
+
+For example, Pulsar SQL (Trino/Presto) will be moved from the main repository to a separate repository. This change will offer the following benefits:
+- A significant reduction in the size of the TGZ and Docker image, saving approximately 400MB.
+- Reduced build time.
+
+Docker image vulnerability scans will also start soon. 
```

Comment:
Could we get some more interesting improvements highlighted?

Perhaps we should add more coverage here based on the recently made PIPs and what has already been made available in the upcoming Pulsar 3.2.0 release?

One of such is PIP-322: Pulsar Rate Limiting Refactoring, https://github.com/apache/pulsar/blob/master/pip/pip-322.md . 
I covered the background in my blog post ["Delivering Improvements in Apache Pulsar for Messaging-as-a-Service Platform Teams"](https://codingthestreams.com/pulsar/2023/11/22/pulsar-slos-and-rate-limiting.html#delivering-improvements-in-apache-pulsar-for-messaging-as-a-service-platform-teams). 

Some possible content:

> An increasing number of messaging-as-a-service platform teams are adopting Apache Pulsar as their main building block for providing messaging services across their organizations.  This is clear validation that the value of Apache Pulsar’s truly multi-tenant architecture is delivering results, making Apache Pulsar a cost-efficient and reliable solution for messaging-as-a-service platform teams in very demanding application environments.
> 
> In the Apache Pulsar project, we are committed to delivering further improvements to the existing multi-tenancy features. One area of improvement is the service level management and capacity management of a large Pulsar system. This is also a key concern of messaging-as-a-service platform teams.
> 
> In December 2023, PIP-322 Pulsar Rate Limiting Refactoring was accepted and completed and will be release as part of Pulsar 3.2.0 release. Rate limiters act as a conduit to more extensive capacity management and Quality of Service (QoS) controls in Pulsar. They are integral to Pulsar's core multi-tenancy features. This refactoring will pave the way for future enhancements in this area.
> 

This contains opinions. :) Is this something that we agree on in the Apache Pulsar project? 
It would be great to have something more interesting than dropping Pulsar SQL support...

Reply:
Thanks @lhotari 
I added your suggestion and also mentioned the Enhanced OTel-based metric system, given that there are ongoing developments

Reply:
Thanks @jak78